x-pack/plugins/data_visualizer/public/application/index_data_visualizer/index_data_visualizer.tsx

```diff
@@ -80,7 +82,80 @@ export const getLocatorParams = (params: {
   return locatorParams;
 };
 
-export const DataVisualizerStateContextProvider: FC<DataVisualizerStateContextProviderProps> = ({
+const DataVisualizerESQLStateContextProvider = () => {
+  const history = useHistory();
+  const { search: urlSearchString } = useLocation();
+
+  const setUrlState: SetUrlState = useCallback(
```

Comment:
We already have a package https://github.com/elastic/kibana/blob/5b1b15af7a9cd478652d996f5e1648ea5cfd4f43/x-pack/packages/ml/url_state/src/url_state.tsx#L103 . Could you please reuse it?